infrastructure/code/convert.go

```diff
@@ -394,6 +394,16 @@ func (s *SarifConverter) toIssues(baseDir string) (issues []snyk.Issue, err erro
 			}
 
 			d.IsIgnored, d.IgnoreDetails = s.getIgnoreDetails(result)
+			additionalDataDetails := getDetailsHtml(d)
+
+			codeIssueData, ok := d.AdditionalData.(snyk.CodeIssueData)
+			if !ok {
+				log.Error().Msg("Failed to cast additional data to CodeIssueData")
+				return issues, errors.New("failed to cast additional data to CodeIssueData")
+			}
+			codeIssueData.Details = additionalDataDetails
+			d.AdditionalData = codeIssueData
+
```

Comment:
Why not something like this instead:
```suggestion

			additionalData.Details = getDetailsHtml(d)
			d.AdditionalData = additionalData
```

Reply:
Also, we should add a check for this in `TestSnykCodeBackendService_convert_shouldConvertIssues` in `conver_test.go` (I should have added one for the ignore details but forgot oops):
```go
	assert.Nil(t, issues[0].IgnoreDetails)
	assert.False(t, issues[0].IsIgnored)
	assert.Contains(t, (issues[0].AdditionalData).(snyk.CodeIssueData).Details, "<h2>Data Flow - 4 steps</h2>")
```

Reply:
I was facing some `panic` errors when I didn't have that type assertion... 🤔  
I removed it 🧹 and all unit tests, as well as testing from IntelliJ and VSCode 💪 
Thank you!